cmd/src/users_prune.go

```diff
@@ -58,65 +60,119 @@ query getCurrentUser {
-				CurrentUser struct {
-					Username string
+			CurrentUser struct {
+				Username string
+			}
+		}
+		if ok, err := cfg.apiClient(apiFlags, flagSet.Output()).NewRequest(currentUserQuery, nil).Do(context.Background(), &currentUserResult); err != nil || !ok {
+			return err
+		}
+
+		// get total users to paginate over
+		totalUsersQuery := `
+query getTotalUsers {
+	site {
+		users {
+			totalCount
+		}
+	}
+}`
+		var totalUsers struct {
+			Site struct {
+				Users struct {
+					TotalCount float64
 				}
 			}
 		}
-		if ok, err := cfg.apiClient(apiFlags, flagSet.Output()).NewRequest(currentUserQuery, nil).DoRaw(context.Background(), &currentUserResult); err != nil || !ok {
+		if ok, err := cfg.apiClient(apiFlags, flagSet.Output()).NewRequest(totalUsersQuery, nil).Do(context.Background(), &totalUsers); err != nil || !ok {
 			return err
 		}
 
+		// get 100 site users
 		getInactiveUsersQuery := `
-query getInactiveUsers {
+		query getInactiveUsers($limit: Int $offset: Int) {
 	site {
 		users {
-			nodes {
+			nodes (limit: $limit offset: $offset) {
+				id
 				username
 				email
 				siteAdmin
 				lastActiveAt
+				deletedAt
 			}
 		}
 	}
 }
 `
 
-		var usersResult struct {
+		// paginate through users
+		var aggregatedUsers struct {
 			Site struct {
 				Users struct {
 					Nodes []SiteUser
 				}
 			}
 		}
 
-		if ok, err := client.NewRequest(getInactiveUsersQuery, nil).Do(ctx, &usersResult); err != nil || !ok {
-			return err
+		// pagination variables, limit set to maximum possible users returned per request
+		offset := 0
+		const limit int = 100
+
+		// paginate requests until all site users have been checked -- this includes soft deleted users
+		for len(aggregatedUsers.Site.Users.Nodes) < int(totalUsers.Site.Users.TotalCount) {
+			pagVars := map[string]interface{}{
+				"offset": offset,
+				"limit":  limit,
+			}
+
+			var usersResult struct {
+				Site struct {
+					Users struct {
+						Nodes []SiteUser
+					}
+					TotalCount float64
+				}
+			}
+			if ok, err := client.NewRequest(getInactiveUsersQuery, pagVars).Do(ctx, &usersResult); err != nil || !ok {
+				return err
+			}
+			// increment graphql request offset by the length of the last user set returned
+			offset = offset + len(usersResult.Site.Users.Nodes)
+			// append graphql user results to aggregated users to be processed against user removal conditions
+			aggregatedUsers.Site.Users.Nodes = append(aggregatedUsers.Site.Users.Nodes, usersResult.Site.Users.Nodes...)
 		}
 
+		// filter users for deletion
 		usersToDelete := make([]UserToDelete, 0)
-		for _, user := range usersResult.Site.Users.Nodes {
+		for _, user := range aggregatedUsers.Site.Users.Nodes {
 			daysSinceLastUse, hasLastActive, err := computeDaysSinceLastUse(user)
 			if err != nil {
 				return err
 			}
 			// never remove user issuing command
-			if user.Username == currentUserResult.Data.CurrentUser.Username {
+			if user.Username == currentUserResult.CurrentUser.Username {
```

Comment:
Nitpick: move that before the call to `computeDaysSinceLastUse`. You don't need to compute anything if you skip the user anyway.

Reply:
better style 💯